src/ordered.rs

```diff
@@ -59,6 +59,7 @@ pub enum Method {
 }
 
 #[cfg(feature = "std")]
+#[cfg_attr(target_family = "wasm", inline(never))]
 fn measure_n_runs(
```

Comment:
This function is what caused the infinite loop. By including the `inline(never)`, the infinite loop stops. This is because the compiler is incorrectly optimizing this function. It assumes that n_runs is a constant. Therefore it will "optimize" (delete) the code that makes the loop work properly. It will be forever running the same instruction over and over again.

Reply:
I understand that the optimizer is wrong here so something is likely broken somewhere, as some of those functions at least the time measurement has a side effect and so should not be skipped, so this bug should be reported somewhere if we manage to get a minimal repro

Reply:
I agree. While I can assure about how the bug happened. I'm still not sure how to replicate it yet. Basically I was logging variables when it was running and suddenly it was working when I logged the variable `n_runs`.

Once we managed to replicate it, maybe we should report it to the wasm32-unknown-unknown compiler community? Since it only happens in wasm.

Reply:
I understand but yes once we have something that looks like it reproes the issue we should find a way to share it with the rust project, worst case they can dispatch it themselves afterwards